decam_process.py

```diff
@@ -15,324 +15,378 @@
+    visit: `str`
+        One or more DECam visit numbers, e.g., `54321`. Multiple visits may be
+        indicated with standard Butler parsing, e.g., `54321^12345`.
+    ccdnum: `str`
+        One or more DECam CCDs (`1` through `62` are allowed). Setting
+        ccdnum='1..62' will process all of the CCDs.
 
     BASH EQUIVALENT:
     $ processCcd.py repo --id visit=visit ccdnum=ccdnum
             --output processedrepo --calib calibrepo
             -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
             -C processccd_config.py
             --config calibrate.doAstrometry=True calibrate.doPhotoCal=True
-    ** to run from bash, 'processccd_config.py' must exist and contain, e.g.,
-        from lsst.meas.algorithms import LoadIndexedReferenceObjectsTask
-        for refObjLoader in (config.calibrate.astromRefObjLoader,
-                            config.calibrate.photoRefObjLoader,
-                            config.charImage.refObjLoader,
-                            ):
-        refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
-        refObjLoader.ref_dataset_name = 'pan-starrs'
-        refObjLoader.filterMap = {"g": "g",
-                                  "r": "r",
-                                  "VR": "g"}
+    ** to run from bash, 'processccd_config.py' must exist and contain
+       all of the refObjLoader information in the code below. repo must also
+       already contain the ref_cats (this is done during doIngest).
 
     RESULT:
     processedrepo/visit populated with subdirectories containing the
     usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR).
     '''
-    if not os.path.isdir(processedrepo):
-        os.mkdir(processedrepo)
-    print('Running ProcessCcd...')
-    OBS_DECAM_DIR = getPackageDir('obs_decam')
-    config = ProcessCcdConfig()
-    # Astrometry retarget party
-    for refObjLoader in (config.calibrate.astromRefObjLoader,
-                         config.calibrate.photoRefObjLoader,
-                         config.charImage.refObjLoader,):
-        refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
-        refObjLoader.ref_dataset_name = 'pan-starrs'  # options are gaia, pan-starrs, sdss
-        refObjLoader.filterMap = {"g": "g",  # 'phot_g_mean_mag' is gaia-specific
-                                  "r": "r",  # all of 'g,r,i,z,y' are options for pan-starrs
-                                  "VR": "g"}
-    args = [repo, '--id', 'visit=' + visit, 'ccdnum=' + ccdnum,
-            '--output', processedrepo,
-            '--calib', calibrepo,
-            '-C', OBS_DECAM_DIR + '/config/processCcdCpIsr.py',
-            '--config', 'calibrate.doAstrometry=True',
-            'calibrate.doPhotoCal=True']
-    ProcessCcdTask.parseAndRun(args=args, config=config)
+    if os.path.isdir(os.path.join(processedrepo, '0'+visit)):
+        print('ProcessCcd has already been run for visit {0}, skipping...'.format(visit))
```

Comment:
Again, maybe add a `return` statement here and move everything else out of the `else` block.